### ISSUE TITLE
Grid Block - (Layout "Auto") breaks the grid when the parent is "Stacked" or "Row"

### ISSUE THREAD

**Issue by ZafarKamal123** [WordPress/gutenberg#65537]
### Description

Grid block is not working as expected when inserted inside the "Stack" (or "Row") container when the layout type is "auto".

https://github.com/user-attachments/assets/0d6bc829-d677-414d-a11e-6dd71f099b26

### Step-by-step reproduction instructions

1. Navigate to any page/post block editor.
2. Insert a "Stack" block.
3. Add a grid block inside the stacked block.
4. Set the "Layout" to "auto" in grid.

### Screenshots, screen recording, code snippet

Here's a simple layout that does not work as expected:

```
<!-- wp:group {"layout":{"type":"flex","flexWrap":"nowrap","orientation":"horizontal","justifyContent":"left"}} -->
<div class="wp-block-group"><!-- wp:group {"style":{"spacing":{"blockGap":"0"}},"layout":{"type":"grid","columnCount":null,"minimumColumnWidth":"12rem"}} -->
<div class="wp-block-group"><!-- wp:paragraph {"align":"center","style":{"elements":{"link":{"color":{"text":"var:preset|color|base"}}}},"backgroundColor":"contrast","textColor":"base"} -->
<p class="has-text-align-center has-base-color has-contrast-background-color has-text-color has-background has-link-color">1</p>
<!-- /wp:paragraph -->

<!-- wp:paragraph {"align":"center","style":{"elements":{"link":{"color":{"text":"var:preset|color|base"}}}},"backgroundColor":"contrast","textColor":"base"} -->
<p class="has-text-align-center has-base-color has-contrast-background-color has-text-color has-background has-link-color">2</p>
<!-- /wp:paragraph -->

<!-- wp:paragraph {"align":"center","style":{"elements":{"link":{"color":{"text":"var:preset|color|base"}}}},"backgroundColor":"contrast","textColor":"base"} -->
<p class="has-text-align-center has-base-color has-contrast-background-color has-text-color has-background has-link-color">3</p>
<!-- /wp:paragraph --></div>
<!-- /wp:group --></div>
<!-- /wp:group -->
```

### Environment info

- WordPress version: **6.6.2**
- Gutenberg Plugin: **19.2.0**
- Browser: **Chrome**
- OS: **MacOS**

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by ndiego** [comment]
I believe the underlying issues here are related to CSS grid elements inside of CSS flex elements. Any thoughts on this @MaggieCabrera

**Comment by MaggieCabrera** [comment]
> I believe the underlying issues here are related to CSS grid elements inside of CSS flex elements. Any thoughts on this @MaggieCabrera

yeah, this is how CSS works, but the user doesn't need to know that. I'm not sure about a workaround for this... Adding width controls to the grid container would help, but I don't know how we want to surface this in the UI. Check this fiddle for the problem and the potential solution:

https://jsfiddle.net/elainemarley85/dabztjL4/2/

**Comment by graylaurenm** [comment]
Do we need the `container-type: inline-size;` set?

**Comment by tellthemachines** [comment]
I'm afraid this is a side effect of the styles on each of the containers being fairly unopinionated, which is useful for flexibility. 

The most straightforward workaround available is to set the Grid to "grow" in the sidebar, under Styles > Width:

<img width="1124" height="202" alt="Image" src="https://github.com/user-attachments/assets/f1dea38f-4e1d-4648-899f-e2b7b81fd413" />

This will make it take up 100% of the Row width, assuming it's the only item inside it. If you plan to have other items inside the Row, you can adjust the relative width of each of them with those controls.

I'm going to go ahead and close this one for now because we're not planning on making any changes to the styles of the actual containers. Thanks for the discussion everyone!
